tests/aws/services/events/test_events_tags.py

```diff
@@ -0,0 +1,289 @@
+        ResourceARN=resource_arn,
+        Tags=[
+            {
+                "Key": "tag1",
+                "Value": "value1",
+            },
+            {
+                "Key": tag_key_2,
+                "Value": "value2",
+            },
+        ],
+    )
+    snapshot.match("tag_resource", response_tag_resource)
+
+    response = aws_client.events.list_tags_for_resource(ResourceARN=resource_arn)
+    snapshot.match("list_tags_for_resource", response)
+
+    response_untag_resource = aws_client.events.untag_resource(
+        ResourceARN=resource_arn,
+        TagKeys=[tag_key_2],
+    )
+    snapshot.match("untag_resource", response_untag_resource)
+
+    response = aws_client.events.list_tags_for_resource(ResourceARN=resource_arn)
+    snapshot.match("list_tags_for_untagged_resource", response)
+
+    response_untag_resource_not_existing_tag = aws_client.events.untag_resource(
+        ResourceARN=resource_arn,
+        TagKeys=[f"not_existing_tag-{short_uid()}"],
+    )
+    snapshot.match("untag_resource_not_existing_tag", response_untag_resource_not_existing_tag)
+
+
+@markers.aws.validated
+@pytest.mark.skipif(
+    not is_v2_provider(),
+    reason="V1 provider does not support this feature",
+)
+@pytest.mark.parametrize("resource_to_tag", ["not_existing_rule", "not_existing_event_bus"])
+def tests_tag_list_untag_not_existing_resource(
+    resource_to_tag,
+    region_name,
+    account_id,
+    aws_client,
+    snapshot,
+):
+    resource_name = short_uid()
+    if resource_to_tag == "not_existing_rule":
+        resource_arn = f"arn:aws:events:{region_name}:{account_id}:rule/{resource_name}"
+    if resource_to_tag == "not_existing_event_bus":
+        resource_arn = f"arn:aws:events:{region_name}:{account_id}:event-bus/{resource_name}"
+
+    tag_key_1 = "tag1"
+    with pytest.raises(aws_client.events.exceptions.ResourceNotFoundException) as error:
+        aws_client.events.tag_resource(
+            ResourceARN=resource_arn,
+            Tags=[
+                {
+                    "Key": tag_key_1,
+                    "Value": "value1",
+                },
+            ],
+        )
+
+    snapshot.match("tag_not_existing_resource_error", error)
+
+    snapshot.add_transformer(
+        snapshot.transform.regex(resource_name, "<not-existing-resource-name>")
+    )
+    with pytest.raises(aws_client.events.exceptions.ResourceNotFoundException) as error:
+        aws_client.events.list_tags_for_resource(ResourceARN=resource_arn)
+    snapshot.match("list_tags_for_not_existing_resource_error", error)
+
+    with pytest.raises(aws_client.events.exceptions.ResourceNotFoundException) as error:
+        aws_client.events.untag_resource(
+            ResourceARN=resource_arn,
+            TagKeys=[tag_key_1],
+        )
+    snapshot.match("untag_not_existing_resource_error", error)
+
+
+@markers.aws.validated
+@pytest.mark.parametrize("event_bus_name", ["event_bus_default", "event_bus_custom"])
```

Comment:
test design note to consider for the future: There's always a trade-off between test complexity and code duplication. That looks like an example to consider whether parametrization is worth the extra complexity given the conditionals cover most of the test. Sometimes duplication or a helper for asserts can more appropriate.